piksi_pps_sync/install/install.sh

```diff
@@ -1,43 +1,80 @@
+  sudo sh -c "echo '# Use USB hotplugging to add new USB devices automatically to the daemon.' >> /etc/default/gpsd"
+  sudo sh -c "echo 'USBAUTO=\"false\"' >> /etc/default/gpsd"
+  sudo sh -c "echo '# Devices gpsd should collect to at boot time.' >> /etc/default/gpsd"
+  sudo sh -c "echo '# They need to be read/writeable, either by user gpsd or the group dialout.' >> /etc/default/gpsd"
+  sudo sh -c "echo 'DEVICES=\"${DEVICE}\"' >> /etc/default/gpsd"
+  sudo sh -c "echo '# Other options you want to pass to gpsd.' >> /etc/default/gpsd"
+  sudo sh -c "echo 'GPSD_OPTIONS=\"-n -r\"' >> /etc/default/gpsd"
+  sudo sh -c "echo '' >> /etc/default/gpsd"
+  sudo sh -c "echo 'GPSD_SOCKET=\"/var/run/gpsd.sock\"' >> /etc/default/gpsd"
+  sudo chmod +x /etc/rc.local
+fi
 
-sudo sh -c "echo '#!/bin/bash' >> /etc/rc.local"
-sudo sh -c "echo '' >> /etc/rc.local"
-sudo sh -c "echo '# Setting GPS UART' >> /etc/rc.local"
-sudo sh -c "echo 'stty -F ${DEVICE} ${BAUD}' >> /etc/rc.local"
-sudo sh -c "echo '# Start GPSD' >> /etc/rc.local"
-sudo sh -c "echo 'service gpsd start' >> /etc/rc.local"
-sudo sh -c "echo '' >> /etc/rc.local"
-sudo sh -c "echo 'exit 0' >> /etc/rc.local"
+sudo dpkg-reconfigure gpsd
 
-sudo chmod +x /etc/rc.local
+echo "Do you wish to create and overwrite a new /etc/rc.local? [y or Y to accept]"
+read create_rc_local
+if [[ $create_rc_local == "Y" || $create_rc_local == "y" ]]; then
+  echo "Creating /etc/rc.local"
+  sudo rm /etc/rc.local
+  sudo sh -c "echo '#!/bin/bash' >> /etc/rc.local"
+  sudo sh -c "echo '' >> /etc/rc.local"
+  sudo sh -c "echo '# Start pps-gpio-modprobe module.' >> /etc/rc.local"
+  sudo sh -c "echo 'modprobe pps-gpio-modprobe gpio=${GPIO_PIN}' >> /etc/rc.local"
+  sudo sh -c "echo '# Setting GPS UART' >> /etc/rc.local"
+  sudo sh -c "echo 'service gpsd stop' >> /etc/rc.local"
+  sudo sh -c "echo 'stty -F ${DEVICE} ${BAUD}' >> /etc/rc.local"
+  sudo sh -c "echo '# Start GPSD' >> /etc/rc.local"
+  sudo sh -c "echo 'service gpsd start' >> /etc/rc.local"
+  sudo sh -c "echo '' >> /etc/rc.local"
+  sudo sh -c "echo 'exit 0' >> /etc/rc.local"
+  sudo chmod +x /etc/rc.local
+fi
 
 # Install chrony.
 sudo apt install chrony -y
 
-sudo sh -c "echo '' >> /etc/chrony/chrony.conf"
-sudo sh -c "echo '# GPS + PPS' >> /etc/chrony/chrony.conf"
-sudo sh -c "echo 'refclock PPS /dev/pps0 lock NMEA' >> /etc/chrony/chrony.conf"
-sudo sh -c "echo 'refclock SHM 0 offset 0.02 delay 0.2 refid NMEA' >> /etc/chrony/chrony.conf"
+echo "Do you wish to append a new PPS refclock to /etc/chrony/chrony.conf? [y or Y to accept]"
+read append_chrony_conf
+if [[ $append_chrony_conf == "Y" || $append_chrony_conf == "y" ]]; then
+  sudo sh -c "echo '' >> /etc/chrony/chrony.conf"
+  sudo sh -c "echo '# GPS + PPS' >> /etc/chrony/chrony.conf"
+  sudo sh -c "echo 'refclock PPS /dev/pps0 lock NMEA' >> /etc/chrony/chrony.conf"
+  sudo sh -c "echo 'refclock SHM 0 delay 0.2 refid NMEA' >> /etc/chrony/chrony.conf"
```

Comment:
out of interest, what does the delay param specify?

Reply:
nevermind, looked it up :-)  
https://wiki.alpinelinux.org/wiki/Chrony_and_GPSD